### ISSUE TITLE
Parametrized tests handling

### ISSUE THREAD

**Issue by botsman** [kavod-io/vscode-jest-test-adapter#26]
Hi!

As far as I can see, parametrized tests are not handled properly.

I am not quite sure if it is possible to explore all the test cases without executing the code though.
But maybe you could take a look at such a case.

Here is an example from the official jest documentation:
```
describe.each([
  [1, 1, 2],
  [1, 2, 3],
  [2, 1, 3],
])('.add(%i, %i)', (a, b, expected) => {
  test(`returns ${expected}`, () => {
    expect(a + b).toBe(expected);
  });

  test(`returned value not be greater than ${expected}`, () => {
    expect(a + b).not.toBeGreaterThan(expected);
  });

  test(`returned value not be less than ${expected}`, () => {
    expect(a + b).not.toBeLessThan(expected);
  });
});
```

These tests are displayed as a single test in the Test Explorer, not as a separate test case for every set of parameters:
<img width="1146" alt="image" src="https://user-images.githubusercontent.com/11509664/79306187-77cd7280-7efd-11ea-820d-45377cb7729b.png">

Also, if I try to run either the whole test or a single test case, nothing happens: neither fail nor success.

Thank you for your work!

**Comment by rossknudsen** [comment]
Cool, sounds like an interesting case to handle.  And thanks for providing an example.

**Comment by rossknudsen** [comment]
I have a feeling that this issue will be related to jest-community/jest-editor-support#29.  We use `jest-editor-support` to parse the test files.  There is an open PR jest-community/jest-editor-support#30 that I think will fix this issue.  Maybe go and add your support for it there?

**Comment by pbotsman** [comment]
Yes, it looks like my issue is related to that in the `jest-editor-support` repository.
Closing this issue then as related to a 3rd party.

Thanks!

**Comment by botsman** [comment]
@rossknudsen hi!

Can I ask you to update to the latest version of https://github.com/jest-community/jest-editor-support ?
It seems that the problem is resolved in the latest version.

Thanks!

**Comment by rossknudsen** [comment]
@botsman would you mind checking how much of the issues that you raised in this ticket are covered by the recent release of v0.8.0 to the marketplace?  If you feel that this ticket has been covered, then please feel free to close.  If there are some minor defects remaining, then you can either document here, or raise them in a separate issue (I'm easy either way).

**Comment by botsman** [comment]
I've checked the latest version with the extension with the base example from 1st message:  
1. In order to discover tests, I need to run them first.  
Before running:
<img width="446" alt="image" src="https://user-images.githubusercontent.com/11509664/105012478-ddc7c700-5a46-11eb-93c6-63a477f528ba.png">
After running:  
<img width="448" alt="image" src="https://user-images.githubusercontent.com/11509664/105012528-e91af280-5a46-11eb-9c97-00017dd1b0ff.png">
If tests are intended to be explored only after execution, then it is fine.



2. After tests are discovered, the first part of parametrized tests remains in the tests UI:
![image](https://user-images.githubusercontent.com/11509664/105012937-65add100-5a47-11eb-8f3b-2795d4024f57.png)


Both of these issues are really minor for me.

Otherwise it looks good, great job guys!
Thank you!

**Comment by clabough** [comment]
This is not working for me. Perhaps because I have nested describes? I have a parent describe that uses beforeEach to set things up for individual tests and I also have some parameterized tests in that describe like 

```
describe.each([[' '], ['  '], [' - '], [', '], ["'"], [":'"]])(
    'inputValues setter should single quote the separator value, when it starts or ends with whitespace or single quote',
    (actualValue: string) => {
      it(` "${actualValue}"`, () => {
        component.inputValues = {separator: actualValue};
        expect(component.separatorInput).toEqual(`'${actualValue}'`);
      });
    }
  );
```

In this case, the Text Explorer panel just shows this before and after running the tests.
![image](https://user-images.githubusercontent.com/5217044/112191961-c1b9e080-8bc3-11eb-9a52-6d04a8d91f90.png)

**Comment by douglasg14b** [comment]
Doesn't work for me either, just stays the same after running.

```
test.each([
  {a: 1, b: 1, expected: 2},
  {a: 1, b: 2, expected: 3},
  {a: 2, b: 1, expected: 3},
])('.add($a, $b)', ({a, b, expected}) => {
  expect(a + b).toBe(expected);
});
```

![image](https://user-images.githubusercontent.com/1400380/125366294-ad445400-e32a-11eb-8fe4-9e4f629114f2.png)
